File: 1514-41-2401-longest-nice-subarray/1514-41-2401-longest-nice-subarray.py
class Solution:
    def longestNiceSubarray(self, nums: List[int]) -> int:
        l,ans,mask=0,0,0
        for r in range(len(nums)):
            while mask&nums[r]:
                mask^=nums[l]
                l+=1
            mask^=nums[r]
            ans = max(ans,r-l+1)
        return ans

File: 1514-41-2401-longest-nice-subarray/1514-41-2401-longest-nice-subarray.java
class Solution {
    public int longestNiceSubarray(int[] nums) {
        int l = 0, ans = 0 ,mask =0;
        for(int r = 0 ; r < nums.length;r++){
            while((nums[r] & mask) !=0){
                mask^=nums[l];
                l+=1;
            }
            mask^=nums[r];
            ans=Math.max(ans,r-l+1);
        }
        return ans;
    }
}